build-scripts/publish.js

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2022 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+/**
+ * @fileoverview
+ *
+ * Before publication, add OS restrictions to the graal packages.
+ * They can't be present before publication as it errors out the installs.
+ */
+
+const fs = require('fs/promises');
+const path = require('path');
+const runCommand = require('./run-command');
+
+const packagesDirPath = path.resolve(__dirname, '../packages');
+
+function isPackageVersionPublished(packageName, version) {
+  return fetch(`https://registry.npmjs.org/${encodeURI(packageName)}/${version}`)
+      .then((res) => res.ok);
+}
+
+async function publishPackagesIfNeeded(packageDir) {
+  const packageJsonPath = `${packageDir}/package.json`;
+  try {
+    await fs.stat(packageJsonPath);
+  } catch {
+    return;
+  }
+  const pkgJson = JSON.parse(await fs.readFile(`${packageDir}/package.json`));
+  const isAlreadyPublished = await isPackageVersionPublished(pkgJson.name, pkgJson.version);
+  if (isAlreadyPublished) {
+    console.log('Already published', pkgJson.name, pkgJson.version);
```

Comment:
No it is intentionally a success. If there is an error in the publication scripts that fails the package publish, then this allows the script to be fixed and only the needed packages re-published.

Also, we technically publish to npm on every push event to github. But only when a version is updated does this push actually do anything since most of the time the current version has already been published.